### ISSUE TITLE
Shelly Plus Plug US not working. 

### ISSUE THREAD

**Issue by DelusionalAI** [phiten/mgos32-to-tasmota32#18]
The update fails right away. Here's the debug log. 
```
[
  {
    "ts": 1684803403.185,
    "data": "Connected."
  },
  {
    "ts": 1684803411.442,
    "level": 2,
    "data": "mg_rpc.c:314 schedule.list via WS_in 10.0.17.127:50812 user admin\n"
  },
  {
    "ts": 1684803411.598,
    "level": 2,
    "data": "mg_rpc.c:314 webhook.list via WS_in 10.0.17.127:50812 user admin\n"
  },
  {
    "ts": 1684803414.909,
    "level": 2,
    "data": "mg_rpc.c:314 switch.toggle via WS_in 10.0.17.127:50812 user admin\n"
  },
  {
    "ts": 1684803414.928,
    "level": 2,
    "data": "shelly_notification:161 Status change of switch:0: {\"id\":0,\"apower\":0,\"current\":0,\"output\":false,\"source\":\"WS_in\"}\n"
  },
  {
    "ts": 1684803415.591,
    "level": 2,
    "data": "mg_rpc.c:314 switch.toggle via WS_in 10.0.17.127:50812 user admin\n"
  },
  {
    "ts": 1684803415.609,
    "level": 2,
    "data": "shelly_notification:161 Status change of switch:0: {\"id\":0,\"output\":true,\"source\":\"WS_in\"}\n"
  },
  {
    "ts": 1684803420.351,
    "level": 2,
    "data": "shelly_notification:161 Status change of switch:0: {\"id\":0,\"aenergy\":{\"by_minute\":[0.000,0.000,0.000],\"minute_ts\":1684803419,\"total\":0.000}}\n"
  },
  {
    "ts": 1684803424.03,
    "level": 2,
    "data": "mg_rpc.c:314 shelly.checkforupdate via WS_in 10.0.17.127:50812 user admin\n"
  },
  {
    "ts": 1684803424.035,
    "level": 2,
    "data": "shelly_update.cpp:170 Checking for updates.\n"
  },
  {
    "ts": 1684803424.053,
    "level": 2,
    "data": "shelly_http.cpp:184 0x3ffdd294 HTTPS GET https://updates.shelly.cloud/update/PlugUS (SSL CA shelly_cloud.pem)\n"
  },
  {
    "ts": 1684803426.545,
    "level": 2,
    "data": "shelly_http.cpp:71 0x3ffdd294 HTTP got 200 OK and 205 byte response\n"
  },
  {
    "ts": 1684803426.564,
    "level": 2,
    "data": "shelly_notification:161 Status change of sys: {\"available_updates\":{\"beta\":{\"version\":\"1.0.0-beta1\"}}}\n"
  },
  {
    "ts": 1684803431.367,
    "level": 2,
    "data": "mg_rpc.c:314 shelly.update via WS_in 10.0.17.127:50812 user admin\n"
  },
  {
    "ts": 1684803431.379,
    "level": 2,
    "data": "shelly_rpc_service.:194 Updating from http://ota.tasmota.com/tasmota32/shelly/mgos32-to-tasmota32-PlusPlugUS.zip\n"
  },
  {
    "ts": 1684803431.389,
    "level": 2,
    "data": "mgos_ota_http_clien:276 Update URL: http://ota.tasmota.com/tasmota32/shelly/mgos32-to-tasmota32-PlusPlugUS.zip\n"
  },
  {
    "ts": 1684803431.547,
    "level": 2,
    "data": "mgos_ota_core.c:269 Starting, timeout 600, commit timeout -1, mem 140268\n"
  },
  {
    "ts": 1684803431.564,
    "level": 2,
    "data": "mgos_ota_core.c:656 FW: PlusPlugUS esp32 v12.5.0 20230505-195325/tasmota-v12.5.0 sig 0x00\n"
  },
  {
    "ts": 1684803431.573,
    "level": 2,
    "data": "shelly_update.cpp:270 incoming fw signatures: 00\n"
  },
  {
    "ts": 1684803431.579,
    "level": 0,
    "data": "shelly_update.cpp:262 Wrong app name 'PlusPlugUS'\n"
  },
  {
    "ts": 1684803431.587,
    "level": 2,
    "data": "mgos_ota_core.c:1051 Update finished, result -13 (Aborted by event handler)\n"
  },
  {
    "ts": 1684803431.613,
    "level": 2,
    "data": "shelly_notification:206 Event from sys: {\"component\":\"sys\", \"event\":\"ota_error\", \"msg\":\"Aborted by event handler\", \"ts\":1684803431.59}\n"
  }
]
```

[diagnostics-shelly-plugus-debug-log-2.txt](https://github.com/phiten/mgos32-to-tasmota32/files/11538165/diagnostics-shelly-plugus-debug-log-2.txt)

**Comment by spiralfive** [comment]
I am experiencing the same issue - the "Wrong app name" error seemingly being the root cause.

**Comment by phiten** [comment]
Thanks for the report. I will have a look at this in the next few days.

**Comment by phiten** [comment]
Shelly changed the name shortly before the release of the products. I adjusted the script accordingly. Please try it out again. Just make sure to use the new link from the list.

**Comment by spiralfive** [comment]
It looks like that fixed it. Thank you! I was able to access the tasmota-AP and point the plug to my home WiFi, though now I cannot seem to access the device at all. I will have to dig into that more when I have time. 

If it helps anyone - I tried using the URL method at first and that did not seem to work. Or I didn't wait long enough. (http://shellyip/ota?url=link to firmware zip) Using the Shelly web UI and setting the firmware link to the PlusUS zip worked.

**Comment by phiten** [comment]
I think the link method you're referring to only works for the non ESP32 Shelly devices. Only the method described in my repo is tested.

**Comment by spiralfive** [comment]
Ah, that makes sense. I did start out using the non-plus instructions before I realized it was a different device.

I am afraid I may have bricked my plug - I got the AP to pull up and set it to my own WiFi, however I haven't been able to access the device since. I tried to reset it to get back to the AP page (with the 6 button press and unplugging it 6 times) but nothing seems to work. 
~~It does not appear to be on the network, though I do have one device I am not sure what it is appearing in my router. No admin UI appears when I try to access it in a browser.~~ - I discovered the mystery device is another new device I added recently. So the plug isn't connecting to the network and I can't get the AP to broadcast again to access it.
